### PR TITLE
Sanitize HTML in forms accepting markdown.

### LINKED ISSUE #469: Markdown fields escape HTML
I think i confirmed that the backend is making EG ```<haskell> in to &lt;haskell&gt;```

### PR DESCRIPTION
**Pull request by fisx**
Fixes #469.

### CODE REVIEW

**Comment by fisx** [comment]
this is fun!  i found two issues just by making `Document` abstract and following the type errors.  (-:

**Comment by fisx** [comment]
ready for review.  this got slightly bigger than i hoped, but i think we can use this as is for the time being.

File: src/Data/Markdown/HtmlWhiteLists.hs [hunk 1]
```diff
@@ -0,0 +1,982 @@
...
+      "TR",
+      "TT",
+      "U",
+      "UL",
+      "VAR"
+      ],
+
+  "denied": [
+      { "key": "APPLET",
+        "reason": "disallow because allows scripting" },
```
Comment (andorp) [inline src/Data/Markdown/HtmlWhiteLists.hs]: Is is here for documentation purposes?

Comment (fisx) [inline src/Data/Markdown/HtmlWhiteLists.hs]: yes i assume so.  this is just copied verbatim from caja, and i'm only using the white lists.  as i indicate in the haddock of `Markdown` though, it is possible that there are some subtleties in the semantics of these json files that i've missed.

**Pull request merged**